singularity-eos/eos/eos.hpp

```diff
@@ -90,82 +90,18 @@ class IdealGas : public EosBase<IdealGas> {
-  inline
-  void SpecificHeatFromDensityTemperature(ConstRealIndexer &&rhos,
-                                          ConstRealIndexer &&temperatures,
-                                          RealIndexer &&cvs,
-                                          const int num,
-                                          LambdaIndexer &&lambdas) const;
-  template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  inline
-  void SpecificHeatFromDensityInternalEnergy(ConstRealIndexer &&rhos,
-                                             ConstRealIndexer &&sies,
-                                             RealIndexer &&cvs,
-                                             const int num,
-                                             LambdaIndexer &&lambdas) const;
-  template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  inline
-  void BulkModulusFromDensityTemperature(ConstRealIndexer &&rhos,
-                                         ConstRealIndexer &&temperatures,
-                                         RealIndexer &&bmods,
-                                         const int num,
-                                         LambdaIndexer &&lambdas) const;
-  template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  inline
-  void BulkModulusFromDensityInternalEnergy(ConstRealIndexer &&rhos,
-                                            ConstRealIndexer &&sies,
-                                            RealIndexer &&bmods,
-                                            const int num,
-                                            LambdaIndexer &&lambdas) const;
-  template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  inline
-  void GruneisenParamFromDensityTemperature(ConstRealIndexer &&rhos,
-                                            ConstRealIndexer &&temperatures,
-                                            RealIndexer &&gm1s,
-                                            const int num,
-                                            LambdaIndexer &&lambdas) const;
-  template<typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
-  inline
-  void GruneisenParamFromDensityInternalEnergy(ConstRealIndexer &&rhos,
-                                               ConstRealIndexer &&sies,
-                                               RealIndexer &&gm1s,
-                                               const int num,
-                                               LambdaIndexer &&lambdas) const;
-  template<typename RealIndexer, typename LambdaIndexer>
-  inline
-  void FillEos(RealIndexer &&rhos, RealIndexer &&temps, RealIndexer &&energies,
-               RealIndexer &&presses, RealIndexer &&cvs, RealIndexer &&bmods,
-               const int num, const unsigned long output,
-               LambdaIndexer &&lambdas) const;
+  // Vector functions that overload the scalar versions declared here
```

Comment:
could use a macro, I know preprocessor, but it could really decrease SLOC and prevent spelling errors:
```
#define SG_ADD_USINGS(MOD) \
using EosBase<MOD>::TemperatureFromDensityInternalEnergy;\
using EosBase<MOD>::InternalEnergyFromDensityTemperature;\
using EosBase<MOD>::PressureFromDensityTemperature;\
using EosBase<MOD>::PressureFromDensityInternalEnergy;\
using EosBase<MOD>::SpecificHeatFromDensityTemperature;\
using EosBase<MOD>::SpecificHeatFromDensityInternalEnergy;\
using EosBase<MOD>::BulkModulusFromDensityTemperature;\
using EosBase<MOD>::BulkModulusFromDensityInternalEnergy;\
using EosBase<MOD>::GruneisenParamFromDensityTemperature;\
using EosBase<MOD>::GruneisenParamFromDensityInternalEnergy;\
using EosBase<MOD>::FillEos;
```

Reply:
That's a good idea.

I'm not completely _against_ using preprocessor macros ;-) . I just don't like when they sacrifice code readability for minor gains. In this case I think the code maintenance boon is very significant and the code readability isn't sacrificed.